### PR TITLE
Let the taskrunner use multiprocessing.

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
Fixes a very longstanding issue where the async task runner doesn't actually use multiprocessing when available.
Runs taskrunner test suites against both threading and multiprocessing for coverage.
Adds an option to run the task workers with multiprocessing if so desired.

### Reviewer guidance
Tests now seem to be running solidly, and it runs against both threading and multiprocessing across all versions.

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by rtibbles** [comment]
Looks like the test flakiness reproduces with abundance on Travis. Going to set this as a draft for now.

File: kolibri/core/tasks/compat.py [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
+    from multiprocessing import synchronize  # noqa
 
     # Proxy Process to Thread to allow seamless substitution
     from multiprocessing import Process as Thread  # noqa
     from multiprocessing import Event  # noqa
 
     # any variable is local to a process, so this is
     # just a dummy
-    local = object
+    class local(object):
```
Comment (aronasorman) [inline kolibri/core/tasks/compat.py:15]: Interesting, what's the reason for the additional class definition here?

File: kolibri/core/tasks/compat.py [hunk 2]
```diff
@@ -1,19 +1,28 @@
+from kolibri.utils.conf import OPTIONS
+
 try:
+    if not OPTIONS["Tasks"]["WORKER_MULTIPROCESSING"]:
```
Comment (aronasorman) [inline kolibri/core/tasks/compat.py]: I think a clearer setting and env var is something like `USE_WORKER_MULTIPROCESSING`

**Review by aronasorman — COMMENTED** [review]
Looks good, just some minor comments. Will test in a bit

File: kolibri/core/tasks/compat.py (hunk 1 above)
Comment (rtibbles) [reply kolibri/core/tasks/compat.py:15]: I think you can't actually directly instantiate `object` because it's an ABC.

Comment (rtibbles) [reply kolibri/core/tasks/compat.py:15]: Weird - seems to work fine locally though, possible that it caused issues in some Python versions.

File: kolibri/core/tasks/compat.py (hunk 2 above)
Comment (rtibbles) [reply kolibri/core/tasks/compat.py]: Updated.

**Comment by aronasorman** [comment]
Ran a local test:
- can confirm that kolibri in worker multiprocessing mode can still import sucessfully
- confirmed that adding this setting creates a new subprocess:
```
# multiprocessing worker mode
#+RESULTS:
: bash───python─┬─python
:               └─312*[{python}]

# vs in non-multiprocessing

#+RESULTS:
: bash───kolibri-0.15.0.───311*[{kolibri-0.15.0.}]

```

**Review by aronasorman — APPROVED** [review]

**Pull request merged**